thread_pthread.c

```diff
@@ -189,6 +189,101 @@ rb_thread_execute_hooks(rb_event_flag_t event)
     }
 }
 
+static rb_internal_thread_store_destructor *rb_internal_thread_store_destructors = NULL;
+static rb_atomic_t rb_internal_thread_store_slots = 0;
+
+unsigned int
+rb_internal_thread_store_slots_count(void)
+{
+    return rb_internal_thread_store_slots;
+}
+
+bool
+rb_internal_thread_store_create_key(rb_internal_thread_store_key_t *key, rb_internal_thread_store_destructor func)
+{
+    int r;
+    if ((r = pthread_rwlock_wrlock(&rb_internal_thread_event_hooks_rw_lock))) {
+        rb_bug_errno("pthread_rwlock_wrlock", r);
+    }
+
+    rb_internal_thread_store_key_t new_key = (rb_internal_thread_store_key_t)rb_internal_thread_store_slots;
+    rb_internal_thread_store_destructor *old_destructors = rb_internal_thread_store_destructors;
+    rb_internal_thread_store_destructor *destructors = ALLOC_N(rb_internal_thread_store_destructor, rb_internal_thread_store_slots + 1);
+    if (old_destructors) {
+        MEMCPY(destructors, old_destructors, rb_internal_thread_store_destructor, rb_internal_thread_store_slots);
+    }
+    destructors[new_key] = func;
+
+    rb_internal_thread_store_destructors = destructors;
+    ATOMIC_INC(rb_internal_thread_store_slots);
+
+    if (old_destructors) {
+        xfree(old_destructors);
+    }
+
+    // TODO: Would be good to iterate over all the existing threads in the current ractor. But that's tricky.
```

Comment:
What I couldn't figure out is how to have access to `ractor.h`, e.g. `GET_RACTOR()` etc.